lib/tokenizer.ts

```diff
@@ -159,6 +159,20 @@ export const tokenizer = (regexpStr: string): Root => {
           // Match if not followed by.
           } else if (c === '!') {
             group.notFollowedBy = true;
+          } else if (c === '<') {
+            const nameMatch = str.slice(i).match(/(\w*)\>/);
```

Comment:
going forward, let's not depend on regex to parse regex, and instead read character by character until a `>` is found.

I know that part of the code at `util.ts` does use regex, but i do plan on someday changing that.

Reply:
Done. Rewrote it as a loop. It still uses a regex for a character test, but I initially did `while (i < str.length && str[i] !== '>')`, and found that doing in inclusive check (for valid characters only vs. anything other than `>`) made the parser report less helpful results when it encountered errors.

I ended up adding some extra sanity checks to validate that capture group names are 100% valid. For some reason, the presence of [this `else` block](https://github.com/fent/ret.js/pull/43/files#diff-2a5d204f18e090aea322959653b540c8876d71dfacddd6983aee56705ec94bd3R168-R174) causes statement coverage to come up one line short, even though both branches are definitely covered by tests ([see here](https://github.com/fent/ret.js/pull/43/files#diff-c8bdf07b252c462de2929820540b0b9803fcb807121e5cc852431515cf199805R74-R81)).